compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/converters/ZipConverter.java

```
@@ -66,36 +66,41 @@ private static void initFS(URI uri) {
-                                    .sorted(Comparator.reverseOrder())
-                                    .limit(1)
-                                    .map(SortablePath::getPath)
-                                    .collect(Collectors.toList());
-                    if (packageID.version.value.isEmpty() && !packageID.orgName.equals(Names.BUILTIN_ORG)
-                            && !packageID.orgName.equals(Names.ANON_ORG) && pathList.size() > 0) {
-                        packageID.version = new Name(pathList.get(0).toFile().getName());
-                    }
+    public Stream<Path> getLatestVersion(Path path, PackageID packageID) {
+        if (!Files.isDirectory(path)) {
+            return Stream.of();
+        }
+        try {
+            List<Path> pathList = new ArrayList<>();
+            if (packageID != null) {
+                String pkgName = packageID.getName().getValue();
+                pathList = Files.list(path)
+                                .map(SortablePath::new)
+                                .filter(SortablePath::valid)
+                                .filter(sortablePath -> validBaloPath(pkgName, sortablePath))
+                                .sorted(Comparator.reverseOrder())
+                                .limit(1)
+                                .map(SortablePath::getPath)
+                                .collect(Collectors.toList());
+                if (packageID.version.value.isEmpty() && !packageID.orgName.equals(Names.BUILTIN_ORG)
+                        && !packageID.orgName.equals(Names.ANON_ORG) && pathList.size() > 0) {
+                    packageID.version = new Name(pathList.get(0).toFile().getName());
                 }
-                return pathList.stream();
-            } catch (IOException ignore) {
             }
+            return pathList.stream();
+        } catch (IOException ex) {
+            return Stream.of();
```

Again, we are ignoring the exception here. There is no way to identify why the exception is thrown if we ignore and return.

---

This exception is thrown if we face an I/O exception when opening the directory specified when listing the files inside it. Since this is done during dependency resolution, we don't want to throw an exception to the user so instead we return an empty stream.

---

Added a comment with a proper explanation with 6744db0